src/hooks/useAddItem.js

```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react';
+import { db } from '../lib/firebase';
+import { addDoc, collection } from 'firebase/firestore';
+import useFetchItems from './useFetchItems';
+
+const useAddItem = (reference) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [successMessage, setSuccessMessage] = useState('');
+  const [error, setError] = useState(null);
+  const [duplicateItemMessage, setDuplicateItemMessage] = useState('');
+  const { data: items } = useFetchItems();
+
+  const addItem = async (newItem, timeframe, token, lastPurchased = null) => {
+    setIsLoading(true);
+    try {
+      const ListRef = collection(db, 'Lists');
+      let addedDocument;
+      checkDuplication(newItem)
+        ? showErrorMessage()
+        : (addedDocument = await addDoc(ListRef, {
+            itemName: newItem,
+            timeframe: parseInt(timeframe),
+            lastPurchased,
+            token,
+          }));
+      if (addedDocument) {
+        setIsLoading(false);
+        showSuccesMessage();
+        setError(false);
+      }
+    } catch (e) {
+      setError(true);
+      setSuccessMessage(null);
+    }
+  };
+
+  //-------------------------------------------------------------------------------------
+  // HELPER FUNCTIONS
+
```

Comment:
Nice separation and commenting. These helper functions can even go in a different file,